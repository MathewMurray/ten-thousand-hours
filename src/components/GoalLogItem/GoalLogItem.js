import React, {Component} from 'react'
import {Link} from 'react-router-dom'
import {Button} from '../../components/Utils/Utils'
import './GoalLogItem.css'

export default class GoalLogItem extends Component {
    render(){
        const {goal} = this.props

        return (
            <Link to={`/Goals/${goal.id}/logs`} className='GoalLogItem'>
                <div className='GoalLogItem__details'>
                    <div className='GoalLogItem__text'>
                    <h3 className='GoalLogItem__Heading'>{goal.id -1}</h3>
                        <h2 className='GoalLogItem__Heading'>{goal.title}</h2>
                        <p className='GoalLogItem__description'>{(goal.target)} hours</p>
                    </div>
                </div>
                <Button type='Delete'>Delete</Button>
            </Link>
        )
    }
}

// function truncate (text) {
//     const words = text.split(' ')

//     if(words.length > 10){
//         return words.slice(0,10).join(' ')+' ...'
//     }
//     return text
// }