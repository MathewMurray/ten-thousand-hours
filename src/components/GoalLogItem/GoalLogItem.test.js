import React from 'react'
import {shallow} from 'enzyme'
import toJson from 'enzyme-to-json'
import GoalLogItem from './GoalLogItem'


describe(`GoalLogItem component`, () => {
    const goal = {
        id: '1',
        title: 'test goal',
    }
    it('renders the Goal logs for goals',() => {
        const wrapper = shallow(<GoalLogItem goal />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})