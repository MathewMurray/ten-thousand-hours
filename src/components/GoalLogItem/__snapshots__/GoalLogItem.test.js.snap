// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalLogItem component renders the Goal logs for goals 1`] = `
<Link
  className="GoalLogItem"
  to="/Goals/undefined/logs"
>
  <div
    className="GoalLogItem__details"
  >
    <div
      className="GoalLogItem__text"
    >
      <h2
        className="GoalLogItem__Heading"
      />
      <p
        className="GoalLogItem__description"
      >
         hours
      </p>
    </div>
  </div>
</Link>
`;
